web/src/features/charts/tooltipCalculations.ts

```diff
@@ -46,6 +47,10 @@ export function getProductionTooltipData(
 
   const generationTypeCapacity = capacity ? capacity[selectedLayerKey] : undefined;
   const generationTypeProduction = production[generationType];
+  const capacitySource =
+    capacitySources && generationTypeCapacity
+      ? capacitySources[selectedLayerKey]
+      : undefined;
```

Comment:
This can be simplified a bit by using optional chaining since we wouldn't need to check the capacity sources object.